pkg/proxy/iptables/proxier_test.go

```diff
@@ -2165,6 +2165,204 @@ COMMIT
 	assertIPTablesRulesEqual(t, expected, fp.iptablesData.String())
 }
 
+func TestOnlyLocalLoadBalancingDetectLocalByInterfacePrefix(t *testing.T) {
```

Comment:
I'm not sure these tests add anything; the tests in `pkg/proxy/util/iptables/traffic_test.go` test that the new detectors generate the right rules, and the existing tests here in `proxier_test.go` test that, whatever rules the localdetector outputs, they get added to the iptables output as expected. So there's no need for extra per-localdetector tests here.

Reply:
hmm you make a good point, Its true that nothing is extra in these tests expect its give the whole picture which the other two tests like you mention do anyways. I am happy to remove the duplicate tests.